core/templates/pages/admin-page/activities-tab/admin-dev-mode-activities-tab.component.ts

```diff
@@ -0,0 +1,213 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the activities tab in the admin panel when Oppia
+ * is in developer mode.
+ */
+
+import { Component, Output, OnInit, EventEmitter } from '@angular/core';
+
+import { AdminDataService } from 'pages/admin-page/services/admin-data.service';
+import { AdminTaskManagerService } from 'pages/admin-page/services/admin-task-manager.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { AdminBackendApiService } from 'domain/admin/admin-backend-api.service';
+
+@Component({
+  selector: 'admin-dev-mode-activities-tab',
+  templateUrl: './admin-dev-mode-activities-tab.component.html',
+})
+export class AdminDevModeActivitiesTabComponent implements OnInit {
+  @Output() setStatusMessage = new EventEmitter<string>();
+  reloadingAllExplorationPossible: boolean = false;
+  demoExplorationIds: string[] = [];
+  numDummyExpsToPublish: number = 0;
+  numDummyExpsToGenerate: number = 0;
+  DEMO_COLLECTIONS: string[][] = [[]];
+  DEMO_EXPLORATIONS: string[][] = [[]];
+
+  constructor(
+    private adminBackendApiService: AdminBackendApiService,
+    private adminDataService: AdminDataService,
+    private adminTaskManagerService: AdminTaskManagerService,
+    private windowRef: WindowRef
+  ) {}
+
+  reloadExploration(explorationId: string): void {
+    if (this.adminTaskManagerService.isTaskRunning()) {
+      return;
+    }
+    if (!this.windowRef.nativeWindow.confirm(
+      'This action is irreversible. Are you sure?')) {
+      return;
+    }
+
+    this.setStatusMessage.emit('Processing...');
+
+    this.adminTaskManagerService.startTask();
+
+    this.adminBackendApiService.reloadExplorationAsync(
+      explorationId
+    ).then(() => {
+      this.setStatusMessage.emit('Data reloaded successfully.');
+      this.adminTaskManagerService.finishTask();
+    }, (errorResponse) => {
+      this.setStatusMessage.emit(
+        'Server error: ' + errorResponse.data.error);
+      this.adminTaskManagerService.finishTask();
+    });
+  }
+
+  printResult(numSucceeded: number, numFailed: number, numTried: number): void {
```

Comment:
I'm assuming all the variable names and functions in these files are the same as the old file, can you confirm?

Reply:
Yes, all the variable names and functions in these files are the same as old.